api.md

```diff
@@ -950,6 +1000,53 @@ SELECT device_id, time_bucket('5 minutes', time) as interval,
  alternative to an `ORDER BY time DESC LIMIT 1` clause to find the
  latest value (which will use indexes).
 
+## locf() [](locf)
+
+The `locf` function allows you to carry the last seen value in an aggregation group forward.
+It can only be used in an aggregation query with `time_bucket_gapfill`.
+The `locf` function call cannot be nested inside other function calls.
+
+#### Required Arguments [](locf-required-arguments)
+
+|Name|Description|
+|---|---|
+| `value` | The value to carry forward (anyelement) |
+
+#### Optional Arguments [](locf-optional-arguments)
+
+|Name|Description|
+|---|---|
+| `prev` | The lookup expression for values outside of the gapfill time range (anyelement) |
```

Comment:
you had (record) here instead of (anyelement) in interpolate. I'm not sure I completely understand what (record) means.

Reply:
record is a composite datatype, since interpolate needs a time:value pair to properly interpolate missing values, while locf takes the previous value verbatim. that's also why locf works with any datatype while interpolate only works with a subset